create_instances_module.rb

```diff
@@ -43,6 +44,19 @@ def add_game
     @games << complete_game
   end
 
+  def add_music_album
+    p 'Is the album in Spotify (true or false)?'
+    spotify = gets.chomp
+
+    p 'Add publish date (DD-MM-YYYY):'
+    album_publish_date = valid_date_input
+
+    base_album = MusicAlbum.new(album_publish_date, spotify)
```

Comment:
Great job with this method my friend, it's almost perfect.
If you look closely at your `MusicAlbum` class you can notice that first, you ask for `spotify` and then the publish date.

Please change the position of what you are passing to the `MusicAlbum.new()` constructor (ex: `MusicAlbum.new(spotify, album_publish_date)`)